src/frontend/XAPI/XAPIStatement.ts

```diff
@@ -107,15 +146,17 @@ export class XAPIStatement {
   }
 
   played(resultExtensions: PlayedResultExtensions): void {
-    const data = {
+    const time: number = XAPIStatement.toFixed(resultExtensions.time);
+    this.startSegments.push(time);
```

Comment:
Can you add an explanation of why you need to keep track of segments inside the XAPI class? Is this really necessary? I would be more comfortable without a stateful class here